tests/eru.Application.Tests/Substitutions/Commands/UploadSubstitutionsCommandTests.cs

```diff
@@ -86,15 +88,25 @@ public async Task DoesUploadWorksCorrectly()
                         Absent = "sample teacher 3"
                     }, 
                 },
-                UploadDateTime = MockData.CorrectDate
+                UploadDateTime = MockData.CorrectDate,
+                SubstitutionsDate = MockData.CorrectDate
             };
             var fakeDbContext = new FakeDbContext();
             var hangfireWrapper = new Mock<IHangfireWrapper>();
             var backgroundJobClient = new Mock<IBackgroundJobClient>();
             hangfireWrapper.Setup(x => x.BackgroundJobClient).Returns(backgroundJobClient.Object);
             var sampleClient = new Mock<IPlatformClient>();
             sampleClient.Setup(x => x.PlatformId).Returns("DebugMessageService");
-            var handler = new UploadSubstitutionsCommandHandler(fakeDbContext, new []{sampleClient.Object}, hangfireWrapper.Object, new ClassParser());
+
+            var classParser = new Mock<IClassesParser>();
+            classParser.Setup(classparser => classparser.Parse(It.IsAny<IEnumerable<string>>())).Returns(
+                (IEnumerable<string> x) =>
+                {
+                    if(x.Contains("I a") & x.Contains("III c")) return Task.FromResult(new[] {new Class(1, "a"), new Class(3, "c")}.AsEnumerable());
```

Comment:
Why is there .AsEnumerable() ? Doesn't array have IEnumerable interface?

Reply:
For some reason, using array as IEnumerable in Task.FromResult does not work. Using Task.FromResult can be avoided by using .ReturnsAsync() method (in Mock setup), but since this method runs synchronously I decided to leave it written in this way.